### PR TITLE
Refactor evaluation: remove module & support qca generator

### PR DESCRIPTION
**Pull request by wwxxzz**

### CODE REVIEW

File: src/pai_rag/evaluation/generator/rag_qca_sample.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        description="The predicted (ground-truth) answer to the example.",
+    )
+    predicted_answer_by: Optional[CreatedBy] = Field(
+        default=None, description="What generated the predicted answer."
+    )
+
+    @property
+    def class_name(self) -> str:
+        """Data example class name."""
+        return "RagQCADataset"
```
Comment (moria97) [inline src/pai_rag/evaluation/generator/rag_qca_sample.py:103]: class_name没有改

File: src/pai_rag/evaluation/generator/rag_qca_sample.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        default_factory=None,
+        description="The contexts used to generate the predicted answer.",
+    )
+    predicted_node_id: Optional[List[str]] = Field(
+        default_factory=None,
+        description="The node id corresponding to the predicted contexts",
+    )
+    predicted_answer: str = Field(
+        default_factory=str,
+        description="The predicted (ground-truth) answer to the example.",
```
Comment (moria97) [inline src/pai_rag/evaluation/generator/rag_qca_sample.py:94]: "predicted (ground-truth) "

predicted应该不是groundtruth?

File: src/pai_rag/evaluation/generator/rag_qca_sample.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    )
+    predicted_answer_by: Optional[CreatedBy] = Field(
+        default=None, description="What generated the predicted answer."
```
Comment (moria97) [inline src/pai_rag/evaluation/generator/rag_qca_sample.py:97]: Maybe "who generated the predicted answer?"

File: src/pai_rag/evaluation/generator/rag_qca_sample.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        description="The reference (ground-truth) answer to the example.",
+    )
+    reference_answer_by: Optional[CreatedBy] = Field(
+        default=None, description="What generated the reference answer."
+    )
+
+    @property
+    def class_name(self) -> str:
+        """Data example class name."""
+        return "RagQCADataset"
```
Comment (moria97) [inline src/pai_rag/evaluation/generator/rag_qca_sample.py:46]: class_name 需要改一下

File: src/pai_rag/evaluation/generator/rag_qca_sample.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+
+class PredictedRagQCASample(BaseLlamaDataExample):
```
Comment (moria97) [inline src/pai_rag/evaluation/generator/rag_qca_sample.py:49]: 这个感觉可以继承labelledRagQcaSample. 一般类里专有名词命名规则是把大写写成Xxxx，如Rag,这里QCA可以改成Qca

**Review by moria97 — APPROVED** [review]

File: src/pai_rag/evaluation/generator/rag_qca_sample.py (hunk 1 above)
Comment (wwxxzz) [reply src/pai_rag/evaluation/generator/rag_qca_sample.py]: Fixed.

File: src/pai_rag/evaluation/generator/rag_qca_sample.py (hunk 2 above)
Comment (wwxxzz) [reply src/pai_rag/evaluation/generator/rag_qca_sample.py]: Fixed.

File: src/pai_rag/evaluation/generator/rag_qca_sample.py (hunk 3 above)
Comment (wwxxzz) [reply src/pai_rag/evaluation/generator/rag_qca_sample.py]: Fixed.

File: src/pai_rag/evaluation/generator/rag_qca_sample.py (hunk 4 above)
Comment (wwxxzz) [reply src/pai_rag/evaluation/generator/rag_qca_sample.py]: Fixed.

File: src/pai_rag/evaluation/generator/rag_qca_sample.py (hunk 5 above)
Comment (wwxxzz) [reply src/pai_rag/evaluation/generator/rag_qca_sample.py]: Fixed.

**Pull request merged**